src/main/java/com/google/sps/data/Section.java

```diff
@@ -15,33 +15,76 @@
-    private String professor;
-    // Meeting times will be sorted in ascending order of time, 
-    // with the first lesson being first in the list
-    private List<TimeRange> meetingTimes;
-
-    public Section(String professor, List<TimeRange> meetingTimes){
-        this.professor = professor;
-        this.meetingTimes = meetingTimes;
-        Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
-
-        for(int i = 0; i < meetingTimes.size() - 1; i++){
-          if(meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
-            throw new IllegalArgumentException("A section's lecture times can't overlap!");
-          }
-        }
+  private String professor;
+  // Meeting times will be sorted in ascending order of time,
+  // with the first lesson being first in the list
+  private List<TimeRange> meetingTimes;
+
+  public Section(String professor, List<TimeRange> meetingTimes) {
+    this.professor = professor;
+    this.meetingTimes = meetingTimes;
+    Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
+
+    for (int i = 0; i < meetingTimes.size() - 1; i++) {
+      if (meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
+        throw new IllegalArgumentException("A section's lecture times can't overlap!");
+      }
     }
+  }
+
+  public List<TimeRange> getMeetingTimes() {
+    return this.meetingTimes;
+  }
+
+  public String getProfessor() {
+    return this.professor;
+  }
+
+  /**
+   * Function to return whether or not a section overlaps with another.
+   * 
+   * @param other The other section.
+   * @return If there's overlap between the current section and the other section
+   */
+  public boolean overlaps(Section other) {
+    LinkedList<TimeRange> thisTimes = new LinkedList<>(this.meetingTimes);
+    LinkedList<TimeRange> otherTimes = new LinkedList<>(other.getMeetingTimes());
+
+    TimeRange thisCurr = thisTimes.removeFirst();
+    TimeRange otherCurr = otherTimes.removeFirst();
 
-    /**
-     * Function to return whether or not a section overlaps with another
-     * @param other
-     * @return If there's overlap between the current section and the other section
-     */
-    public boolean overlaps(Section other){
-        //TODO: This
+    while (!thisTimes.isEmpty() && !otherTimes.isEmpty()) {
+      if (thisCurr.overlaps(otherCurr)) {
         return false;
+      } else if (thisCurr.end() < otherCurr.end()) {
+        thisCurr = thisTimes.removeFirst();
+      } else {
+        otherCurr = otherTimes.removeFirst();
+      }
     }
+
+    // This bit of redundant logic is to account for the single element list case.
+    if (thisCurr.overlaps(otherCurr)) {
+      return true;
+    }
+    return false;
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    } else {
+      return other instanceof Section && equals(this, (Section) other);
+    }
+  }
+
+  private boolean equals(Section a, Section b) {
```

Comment:
This seems like it should be a static method since it doesn't rely on `this` at all